ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -363,4 +391,46 @@ public void testListTablesStatement() throws Exception {
 
     assertEquals(expectedTable, testTables.get(0));
   }
+
+  @Test
+  public void shouldFailForIncorrectCSASStatementResultType() throws Exception {
+    KsqlResource testResource = TestKsqlResourceUtil.get();
+    String ksqlString1 = "CREATE STREAM s1 AS SELECT * FROM test_table;";
+
+    Response response1 = testResource.handleKsqlStatements(new KsqlRequest(ksqlString1, Collections
+        .EMPTY_MAP));
+    KsqlEntityList result1 = (KsqlEntityList) response1.getEntity();
+    assertTrue("Incorrect response size.", result1.size() == 1);
+    assertThat(result1.get(0), instanceOf(ErrorMessageEntity.class));
+    ErrorMessageEntity errorMessageEntity1 = (ErrorMessageEntity) result1.get(0);
+    assertTrue(errorMessageEntity1.getErrorMessage().getMessage().equalsIgnoreCase("Invalid "
```

Comment:
Does the exception thrown by `validateExecutionPlan` eventually boil down to this error message? If so, would it make more sense to test for validity at a more fine grained level, where we can check the typed exception directly?

Reply:
Also, do we have tests for the positive code path, where there are no errors?

Reply:
Yes, but the 'getStatementExecutionPlan' method is private and because of that I have the test in the caller.
The positive code path test needs to be added too as you pointed out but it will include much more that the fix for this regression bug and can be part of another PR that improves the test coverage for this KsqlResource.